### ISSUE TITLE
MapHandler not found exception when using build in Map-feature

### ISSUE THREAD

**Issue by CayasSoftware** [dotnet/maui#12088]
### Description

When trying to use the built-in map functionality of the MAUI framework, a "Handler not found for view Microsoft.Maui.Controls.Maps.Map" error message occurs in Visual Studio for Mac while launching the app.

Dev environment:

Visual Studio Community 2022 for Mac Preview
Version 17.5 Preview (17.5 build 437)
Installation UUID: d18b378b-b58e-4d05-a47a-ba8a0ff0e318

Runtime
.NET 7.0.0 (64-bit)
Architecture: Arm64

Roslyn (Language Service)
4.4.0-3.22461.4+8ab250290a4010c11a21521f78dbc87dbb7aac81

NuGet
Version: 6.3.1.1

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.100/Sdks
SDK Versions:
	7.0.100
	6.0.403
	6.0.402
	6.0.401
	6.0.400
	6.0.301
	6.0.300
	6.0.203
	6.0.202
MSBuild SDKs: /Applications/Visual Studio (Preview).app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET SDK (x64)
SDK Versions:
	3.1.425
	3.1.424
	3.1.423
	3.1.422
	3.1.420
	3.1.419
	3.1.418

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	7.0.0
	6.0.11
	6.0.10
	6.0.9
	6.0.8
	6.0.6
	6.0.5
	6.0.4

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	3.1.31
	3.1.30
	3.1.29
	3.1.28
	3.1.26
	3.1.25
	3.1.24

Xamarin.Profiler
Version: 1.8.0.19
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin.Android
Version: 13.1.0.1 (Visual Studio Community)
Commit: xamarin-android/d17-4/13ba222
Android SDK: /Users/sebastian/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		12.0 (API level 31)
		11.0 (API level 30)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 33.0.2
SDK Build Tools Version: 32.0.0

Build Information: 
Mono: a96bde9
Java.Interop: xamarin/java.interop/d17-4@fcc33ce2
SQLite: xamarin/sqlite/3.39.3@23e1ae7
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@0be567a

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.16.1
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.5.0.8
Hash: 6f27afd
Branch: remotes/origin/HEAD
Build date: 2022-11-02 21:36:54 UTC

Android Device Manager
Version: 0.0.0.1217
Hash: 6175224
Branch: main
Build date: 2022-11-02 21:36:54 UTC

Apple Developer Tools
Xcode 14.2 (21534)
Build 14C18

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 16.0.0.72 (Visual Studio Community)
Hash: 6756a1146
Branch: release/6.0.4xx-xcode14
Build date: 2022-09-21 08:51:06-0400

Xamarin Designer
Version: 17.5.1.13
Hash: 14bdde937f
Branch: remotes/origin/d17-5
Build date: 2022-11-02 21:36:49 UTC

Build Information
Release ID: 1705000437
Git revision: a5d0fb913c1ff781c80fe725a3e0511c34e26a08
Build date: 2022-11-02 21:34:22+00
Build branch: release-17.5
Build lane: release-17.5

Operating System
Mac OS X 12.6.1
Darwin 21.6.0 Darwin Kernel Version 21.6.0
    Thu Sep 29 20:13:56 PDT 2022
    root:xnu-8020.240.7~1/RELEASE_ARM64_T6000 arm64

### Steps to Reproduce

1. Create a new MAUI app with .NET 7
2. Add NuGet package Microsoft.Maui.Controls.Maps with version 7.0.52
3. Add UseMauiMaps() to the builder in MauiProgram.cs
4. Add Map-Control into a Xaml-File
5. Start the project on a simulator (testet iOS 16.1 & 16.2)

### Link to public reproduction project repository

https://github.com/CayasSoftware/maui-maps-error

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS 16.1, iOS 16.2, MacOS 12.16.1

**Comment by jfversluis** [comment]
Thanks for the report @CayasSoftware ! I also already mentioned it on Twitter where you reached out, but just downloading your reproduction sample (thank you by the way!) and running that, works fine for me...

**Comment by CayasSoftware** [comment]
What question or type of information could I provide to move on?

Currently thinking about making a screen recording to prove that it happen. Otherwise would that really help?

My dev environment information missing the Workload versions. I will provide them later.

**Comment by CayasSoftware** [comment]
I created a video for which I created multiple projects that shows the same behavior.
https://user-images.githubusercontent.com/1324824/207802647-82288a54-de12-4403-9eb4-723dc84728c0.mov

**Comment by carolzbnbr** [comment]
Same problem here. 
Using Rider for mac instead of VS for mac, and also having latest MAUI/Net6 updates.

Attempted to downgrade the map nuget package to previous version, but the problem persists.

**Comment by CayasSoftware** [comment]
Tested with a different setup and can confirm that it works there. @carolzbnbr are you on the Visual Studio for Mac Preview channel?

Tested dev environment on which everything works:

Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
maui-ios                   7.0.52/7.0.100         SDK 7.0.100        
maui-android               7.0.52/7.0.100         SDK 7.0.100        
maccatalyst                16.1.1481/7.0.100      SDK 7.0.100        
wasm-tools-net6            7.0.1/7.0.100          SDK 7.0.100


Visual Studio Community 2022 for Mac
Version 17.4.1 (build 28)
Installation UUID: b14066fc-7f4e-4da0-af65-ae39e93ea766

Runtime
.NET 6.0.9 (64-bit)
Architecture: Arm64

Roslyn (Language Service)
4.4.0-6.22578.12+3c6ab8e1715e5b080fb7bb77070810ab71e09387

NuGet
Version: 6.3.1.1

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.100/Sdks
SDK Versions:
    7.0.100
    7.0.100-rc.2.22477.23
    6.0.403
    6.0.402
    6.0.401
    6.0.400
MSBuild SDKs: /Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET SDK (x64)
SDK Versions:
    6.0.403
    6.0.402
    6.0.401
    6.0.400
    6.0.106
    6.0.105
    6.0.104
    6.0.101
    5.0.408
    5.0.407
    5.0.404
    3.1.425
    3.1.424
    3.1.423
    3.1.422
    3.1.420
    3.1.419
    3.1.418
    3.1.416

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
    7.0.0
    7.0.0-rc.2.22472.3
    6.0.11
    6.0.10
    6.0.9
    6.0.8

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
    6.0.11
    6.0.10
    6.0.9
    6.0.8
    6.0.6
    6.0.5
    6.0.4
    6.0.1
    5.0.17
    5.0.16
    5.0.13
    3.1.31
    3.1.30
    3.1.29
    3.1.28
    3.1.26
    3.1.25
    3.1.24
    3.1.22

Xamarin.Profiler
Version: 1.8.0.19
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin.Android
Version: 13.1.0.1 (Visual Studio Community)
Commit: xamarin-android/d17-4/13ba222
Android SDK: /Users/martin/Library/Developer/Xamarin/android-sdk-macosx
    Supported Android versions:
        12.1 (API level 32)
        8.0  (API level 26)
        12.0 (API level 31)
        5.1  (API level 22)
        11.0 (API level 30)
        10.0 (API level 29)
        7.1  (API level 25)
        9.0  (API level 28)
        13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 33.0.3
SDK Build Tools Version: 33.0.0

Build Information: 
Mono: a96bde9
Java.Interop: xamarin/java.interop/d17-4@fcc33ce2
SQLite: xamarin/sqlite/3.39.3@23e1ae7
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@0be567a

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.12
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.4.0.54
Hash: 6eabb9e
Branch: remotes/origin/d17-4
Build date: 2022-12-02 15:36:24 UTC

Android Device Manager
Version: 0.0.0.1206
Hash: 886af39
Branch: 886af39
Build date: 2022-12-02 15:36:24 UTC

Xamarin Designer
Version: 17.4.0.136
Hash: d49c9ff6d3
Branch: remotes/origin/d17-4
Build date: 2022-12-02 15:36:20 UTC

Apple Developer Tools
Xcode 14.2 (21534)
Build 14C18

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 16.0.0.72 (Visual Studio Community)
Hash: 6756a1146
Branch: release/6.0.4xx-xcode14
Build date: 2022-09-21 08:51:06-0400

Build Information
Release ID: 1704010028
Git revision: f17dcd4436244cc06ed14dc5238c4d3fa788144f
Build date: 2022-12-02 15:34:08+00
Build branch: release-17.4
Build lane: release-17.4

Operating System
Mac OS X 12.6.0
Darwin 21.6.0 Darwin Kernel Version 21.6.0
    Mon Aug 22 20:19:52 PDT 2022
    root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64

**Comment by carolzbnbr** [comment]
Hey @CayasSoftware , no, I'm on the stable channel. 

Actually was using net 6 until I joined this thread/issue Here today. 

Now I have migrated to net 7, also on stable channel, but now I have other big issues related to Android 33 itself. 😜

I'm not on the computer right now but as soon as I arrive I will attempt the  preview channel, maybe it might fix my android problem. 

Anyway, thanks for trying to help me. 

I will get back here after I attempt the preview channel. 

Have a nice day.

**Comment by carolzbnbr** [comment]
Hey @CayasSoftware , 

Jus to let you know. 
I was able to run the map without issues on Android using Net7 and Visual Studio for mac (stable channel) having the latest updates installed yesterday.

**Comment by pkzOR** [comment]
I reported this back in Nov: https://github.com/dotnet/maui/issues/11169 and set to  a backlog item.
How are we supposed to proceed?

**Comment by CayasSoftware** [comment]
@pkzOR It looks like your case is different in that way that it seems to be Windows related.

**Comment by cdavidyoung** [comment]
I am having this issue on MacCatalyst.  It happens when I upgrade from Microsoft.Maui.Controls.Maps version 7.0.49 to 7.0.52.  It causes an exception when I try to navigate to the page where the map is displayed.  I added a try/catch and am printing the exception to the console.  I have reverted back to 7.0.49 and my map page now works fine.  BTW, I tested 7.0.52 on Android it it works fine.

        public async Task OnTapGestureRecognizerTapped(object sender, EventArgs e)
        {
           Record rec = (Record)((TappedEventArgs)e).Parameter;
           string jiffyId = rec.ID.ToString();
           string dbName = "DB1";
           if (MyDB == Database2)
              dbName = "DB2";
           try
            {
                await Shell.Current.GoToAsync($"TourPage?jiffyTitle={$"🧳Tour {GD}"}&&jiffyId={jiffyId}&&jiffyDB={dbName}");
            }
            catch(Exception ex)
            {
                Console.WriteLine($"OnTapGestureRecognizerTapped {ex}");
            }
           
        }

2022-12-20 12:13:47.687 CatchAll[48609:5759666] OnTapGestureRecognizerTapped Microsoft.Maui.Platform.HandlerNotFoundException: Handler not found for view Microsoft.Maui.Controls.Maps.Map.
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
2022-12-20 12:13:47.687 CatchAll[48609:5759666]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
2022-12-20 12:13:47.687 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
2022-12-20 12:13:47.687 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
2022-12-20 12:13:47.687 CatchAll[48609:5759666]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
2022-12-20 12:13:47.687 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.ContentViewHandler.UpdateContent(IContentViewHandler handler)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ContentViewHandler.MapContent(IContentViewHandler handler, IContentView page)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IContentViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
   at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ContentViewHandler.SetVirtualView(IView view)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ViewExtensions.ToHandler(IView view, IMauiContext context)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRenderer.PushPage(Page page, Boolean animated, TaskCompletionSource`1 completionSource)
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRenderer.OnPushRequested(NavigationRequestedEventArgs e)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRenderer.OnNavigationRequested(Object sender, NavigationRequestedEventArgs e)
   at Microsoft.Maui.Controls.ShellSection.InvokeNavigationRequest(NavigationRequestedEventArgs args)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.ShellSection.OnPushAsync(Page page, Boolean animated)
   at Microsoft.Maui.Controls.ShellSection.PushStackOfPages(List`1 pages, Nullable`1 animate)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at Microsoft.Maui.Controls.ShellSection.GoToAsync(ShellNavigationRequest request, ShellRouteParameters queryData, IServiceProvider services, Nullable`1 animate, Boolean isRelativePopping)
   at Microsoft.Maui.Controls.ShellNavigationManager.GoToAsync(ShellNavigationParameters shellNavigationParameters, ShellNavigationRequest navigationRequest)
2022-12-20 12:13:47.688 CatchAll[48609:5759666]    at jiffyLog.ViewModels.RecordsViewModel.OnTapGestureRecognizerTapped(Object sender, EventArgs e) in /Users/charlesyoung/Projects/jiffyLog/ViewModels/RecordsViewModel.cs:line 700
